### PR TITLE
blog: add monthly-report (2024.10.01-11.30)

### PR DESCRIPTION
**Pull request by Yilialinn**
Add APISIX community bi-monthly report

### CODE REVIEW

File: blog/en/blog/2024/11/30/monthly-report.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+This PR changed `scope` into `scopes` according to OAuth2/OIDC rules. It also replaced `google-cloud-logging/oauth.lua` with `utils/google-cloud-oauth.lua`.
+
+2. [Add `ai-content-moderation` Plugin](https://github.com/apache/apisix/pull/11541) (Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))
+
+The `ai-prompt-decorator` plugin simplifies access to LLM providers, such as OpenAI and Anthropic, and their models by appending or prepending prompts into the request.
+
+3. [Add `ai-rag` Plugin](https://github.com/apache/apisix/pull/11568)（Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))
+
+Implement the ai-rag plugin that uses RAG (Retrieval Augmented Generation), to return information about untrained events/facts from LLMs.
```
Comment (kayx23) [inline blog/en/blog/2024/11/30/monthly-report.md]:
```suggestion
Implement the `ai-rag` plugin that uses RAG (Retrieval Augmented Generation), to return information about untrained events/facts from LLMs.
```

File: blog/en/blog/2024/11/30/monthly-report.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+As the Grafana dashboard only has panels for RPS per status code and RPS per service/route, this PR also adds a total RPS panel for the overview of RPS changes.
+
+5. [Use `setmetatable` to Set Hidden Variables without Effecting Serialisation in `body-transformer` Plugin](https://github.com/apache/apisix/pull/11770) (Contributor: [Revolyssup](https://github.com/Revolyssup))
+
+Users hope to directly use the decoded body in the template of `body-transformer` plugin, modify its fields, and then encode it again.
+
+6. [Support configuring `key_claim_name` in `jwt-auth` Plugin](https://github.com/apache/apisix/pull/11772) (Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))
+
+Configuring key_claim_name in the JWT plugin is essential for specifying which claim in the JWT contains the key that identifies the secret used for validating the token.
```
Comment (kayx23) [inline blog/en/blog/2024/11/30/monthly-report.md]:
```suggestion
Configuring `key_claim_name` in the JWT plugin is essential for specifying which claim in the JWT contains the key that identifies the secret used for validating the token.
```

File: blog/zh/blog/2024/11/30/monthly-report.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## 贡献者统计
+
+![贡献者名单](https://static.apiseven.com/uploads/2024/11/29/7z0d7q0r_contributors-nov.png)
+
+![新晋贡献者](https://static.apiseven.com/uploads/2024/11/29/53aNHiqa_nov-new-contributors.png)
+
+## 近期亮点功能
+
+1. [重构 google-cloud-logging 文档](https://github.com/apache/apisix/pull/11596)（贡献者：[HuanXin-Chen](https://github.com/HuanXin-Chen))
```
Comment (kayx23) [inline blog/zh/blog/2024/11/30/monthly-report.md]:
```suggestion
1. [重构 `google-cloud-logging` 文档](https://github.com/apache/apisix/pull/11596)（贡献者：[HuanXin-Chen](https://github.com/HuanXin-Chen))
```

File: blog/zh/blog/2024/11/30/monthly-report.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+由于 Grafana 仪表板仅包含按状态码和按服务/路由的 RPS 面板，因此此 PR 还添加了一个总 RPS 面板，用于概览 RPS 的变化情况。
+
+5. [在 `body-transformer` 插件中使用 `setmetatable` 设置隐藏变量而不影响序列化](https://github.com/apache/apisix/pull/11770)（贡献者：[Revolyssup](https://github.com/Revolyssup))
+
+此 PR 能让用户直接在 `body-transformer` 的模板中使用解码后的body，修改其字段，然后再次进行编码。
+
+6. [支持在 `jwt-auth` 插件中配置 `key_claim_name`](https://github.com/apache/apisix/pull/11772)（贡献者：[shreemaan-abhishek](https://github.com/shreemaan-abhishek))
+
+支持在 jwt-auth 插件中配置 `key_claim_name`，用于指定 jwt-auth 中的哪个声明包含用于验证令牌的密钥，从而识别所使用的 secret。
```
Comment (kayx23) [inline blog/zh/blog/2024/11/30/monthly-report.md]:
```suggestion
支持在 `jwt-auth` 插件中配置 `key_claim_name`，用于指定 `jwt-auth` 中的哪个声明包含用于验证令牌的密钥，从而识别所使用的 secret。
```

File: blog/en/blog/2024/11/30/monthly-report.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+## Contributor Statistics
+
+![Apache APISIX Contributors List](https://static.apiseven.com/uploads/2024/11/29/7z0d7q0r_contributors-nov.png)
+
+![Apache APISIX New Contributors](https://static.apiseven.com/uploads/2024/11/29/53aNHiqa_nov-new-contributors.png)
+
+## Recent Feature Highlights
+
+1. [Refactor google-cloud-logging Plugin](https://github.com/apache/apisix/pull/11596) (Contributor: [HuanXin-Chen](https://github.com/HuanXin-Chen))
```
Comment (kayx23) [inline blog/en/blog/2024/11/30/monthly-report.md]:
```suggestion
1. [Refactor `google-cloud-logging` Plugin](https://github.com/apache/apisix/pull/11596) (Contributor: [HuanXin-Chen](https://github.com/HuanXin-Chen))
```

**Comment by yzeng25** [comment]
Should it be CONTRIBUTOR in the title since there is only 1 new contributor?

image

File: blog/en/blog/2024/11/30/monthly-report.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+title: "Monthly Report (October 01 - November 30)"
```
Comment (yzeng25) [inline blog/en/blog/2024/11/30/monthly-report.md]:
```suggestion
title: "Bi-monthly Report (October 01 - November 30)"
```

Comment (yzeng25) [reply blog/en/blog/2024/11/30/monthly-report.md]: file name as well blog/en/blog/2024/11/30/bi-monthly-report.md

File: blog/zh/blog/2024/11/30/monthly-report.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+---
+title: "社区月报 (10.01 - 11.30)"
```
Comment (yzeng25) [inline blog/zh/blog/2024/11/30/monthly-report.md]:
```suggestion
title: "社区双月报 (10.01 - 11.30)"
```

Comment (yzeng25) [reply blog/zh/blog/2024/11/30/monthly-report.md]: file name as well 

blog/zh/blog/2024/11/30/bi-monthly-report.md

File: blog/en/blog/2024/11/30/monthly-report.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+keywords: ["Apache APISIX", "API Gateway", "Monthly Report", "Contributor"]
+description: Our monthly Apache APISIX community report generates insights into the project's monthly developments. The reports provide a pathway into the Apache APISIX community, ensuring that you stay well-informed and actively involved.
+tags: [Community]
+image: https://static.apiseven.com/uploads/2024/11/29/skKTqZaH_nov-cover-en.png
+---
+
+> We have recently made some additions and improvements to specific features within Apache APISIX. The main improvements include the addition of `ai-content-moderation` and `ai-rag` plugins, and a total request panel in the Grafana dashboard among other enhancements. For detailed information, please read the monthly report.
```
Comment (yzeng25) [inline blog/en/blog/2024/11/30/monthly-report.md]:
```suggestion
> We have recently made some additions and improvements to specific features within Apache APISIX. The main improvements include the addition of `ai-content-moderation` and `ai-rag` plugins, and a total request panel in the Grafana dashboard among other enhancements. For detailed information, please read the bi-monthly report.
```

File: blog/zh/blog/2024/11/30/monthly-report.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+keywords: ["Apache APISIX", "API 网关", "社区月报", "贡献者"]
+description: Apache APISIX 社区的月报旨在帮助社区成员更全面地了解社区的最新动态，方便大家参与到 Apache APISIX 社区中来。
+tags: [Community]
+image: https://static.apiseven.com/uploads/2024/11/29/69Av0Gu3_nov-cover-cn.png
+---
+
+> 最近，我们新增并改进了 Apache APISIX 的部分功能，包括新增 `ai-content-moderation` 和 `ai-rag` 插件，在 Grafana 仪表盘中新增总请求变化面板等。有关更多功能新亮点，请阅读本期月报。
```
Comment (yzeng25) [inline blog/zh/blog/2024/11/30/monthly-report.md]:
```suggestion
> 最近，我们新增并改进了 Apache APISIX 的部分功能，包括新增 `ai-content-moderation` 和 `ai-rag` 插件，在 Grafana 仪表盘中新增总请求变化面板等。有关更多功能新亮点，请阅读本期双月报。
```

**Comment by Yilialinn** [comment]
> Should it be CONTRIBUTOR in the title since there is only 1 new contributor?
> 
> image

Fixed

**Review by kayx23 — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**